docs/resources/dcs_redis_instance.md

```diff
@@ -0,0 +1,261 @@
+
+* `engine_version` - (Required, String, ForceNew) Specifies the version of a cache engine.
+  Currently, the valid values are `3.0`, `4.0` and `5.0`. Changing this creates a new instance.
+
+* `capacity` - (Required, Float) The total memory of the cache, in GB.
+  + **Redis4.0 and Redis5.0**: Stand-alone and active/standby type instance values: `0.125`, `0.25`, `0.5`, `1`, `2`,
+    `4`, `8`, `16`, `32` and `64`.
+    Cluster instance specifications support `24`, `32`, `48`, `64`, `96`, `128`, `192`, `256`, `384`, `512`, `768` and
+    `1024`.
+  + **Redis3.0**: Stand-alone and active/standby type instance values: `2`, `4`, `8`, `16`, `32` and `64`.
+    Proxy cluster instance specifications support `64`, `128`, `256`, `512`, and `1024`.
+  + **Memcached**: Stand-alone and active/standby type instance values: `2`, `4`, `8`, `16`, `32` and `64`.
+
+* `resource_spec_code` - (Required, String) The specification code of the cache instance, which including the total
+  memory, available memory, maximum number of connections allowed, maximum/assured bandwidth and reference performance.
+  You can specify specification codes to create different type of Redis instances.
+  It can be obtained through this data source `huaweicloud_dcs_flavors`.
+
+* `available_zones` - (Required, List, ForceNew) The code of the AZ where the cache node resides.
+  Master/Standby, Proxy Cluster, and Redis Cluster DCS instances support cross-AZ deployment.
+  You can specify an AZ for the standby node. When specifying AZs for nodes, use commas (,) to separate AZs.
+  Changing this creates a new instance.
+
+* `vpc_id` - (Required, String, ForceNew) The ID of VPC which the instance belongs to.
+  Changing this creates a new instance resource.
+
+* `subnet_id` - (Required, String, ForceNew) The ID of subnet which the instance belongs to.
+  Changing this creates a new instance resource.
+
+* `security_group_id` - (Optional, String) The ID of the security group which the instance belongs to.
+  This parameter is mandatory for Redis 3.0 version.
+
+* `ip` - (Optional, String, ForceNew) The IP address of the DCS instance, which can only be the currently available IP
+  address the selected subnet.
+  You can specify an available IP for the Redis instance (except for the Redis Cluster type).
+  If omitted, the system will automatically allocate an available IP address to the Redis instance.
+  Changing this creates a new instance resource.
+
+* `port` - (Optional, Int) Port customization, which is supported only by Redis 4.0 and Redis 5.0 instances.
+  Default value is 6379.
+
+* `password` - (Optional, String) Specifies the password of a DCS instance.
+  The password of a DCS Redis instance must meet the following complexity requirements:
+  + Enter a string of 8 to 32 bits in length.
+  + The new password cannot be the same as the old password.
+  + Must contain three combinations of the following four characters: Lower case letters, uppercase letter, digital,
```

Comment:
Move this rule to Line 133 could be better

Reply:
done